Compilation error using boost 1.76 and gcc 11.2.1

## Expected Behavior

HPX master should compile with boost 1.76 and The CXX compiler identification is GNU 11.2.1


## Actual Behavior

```
 33%] Building CXX object libs/core/futures/CMakeFiles/hpx_futures.dir/src/future_data.cpp.o
In file included from /usr/include/boost/lockfree/queue.hpp:15,
                 from /hpx/libs/core/schedulers/include/hpx/schedulers/lockfree_queue_backends.hpp:17,
                 from /hpx/libs/core/schedulers/include/hpx/schedulers/local_priority_queue_scheduler.hpp:18,
                 from /hpx/libs/core/thread_pools/src/scheduled_thread_pool.cpp:8:
/usr/include/boost/lockfree/queue.hpp: In instantiation of 'class boost::lockfree::queue<hpx::threads::thread_id_ref, hpx::util::aligned_allocator<hpx::threads::thread_id_ref> >':
/hpx/libs/core/schedulers/include/hpx/schedulers/lockfree_queue_backends.hpp:92:24:   required from 'struct hpx::threads::policies::lockfree_fifo_backend<hpx::threads::thread_id_ref>'
/hpx/libs/core/schedulers/include/hpx/schedulers/thread_queue.hpp:1167:25:   required from 'class hpx::threads::policies::thread_queue<std::mutex, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo, hpx::threads::policies::lockfree_fifo>'
/hpx/libs/core/schedulers/include/hpx/schedulers/local_priority_queue_scheduler.hpp:1379:27:   required from 'class hpx::threads::policies::local_priority_queue_scheduler<>'
/hpx/libs/core/thread_pools/src/scheduled_thread_pool.cpp:27:29:   required from here
/usr/include/boost/lockfree/queue.hpp:95:5: error: static assertion failed: (boost::has_trivial_destructor<T>::value)
   95 |     BOOST_STATIC_ASSERT((boost::has_trivial_destructor<T>::value));
      |     ^~~~~~~~~~~~~~~~~~~
/usr/include/boost/lockfree/queue.hpp:95:5: note: 'boost::integral_constant<bool, false>::value' evaluates to false
/usr/include/boost/lockfree/queue.hpp:99:5: error: static assertion failed: (boost::has_trivial_assign<T>::value)
   99 |     BOOST_STATIC_ASSERT((boost::has_trivial_assign<T>::value));
      |     ^~~~~~~~~~~~~~~~~~~
/usr/include/boost/lockfree/queue.hpp:99:5: note: 'boost::integral_constant<bool, false>::value' evaluates to false
make[2]: *** [libs/core/thread_pools/CMakeFiles/hpx_thread_pools.dir/build.make:76: libs/core/thread_pools/CMakeFiles/hpx_thread_pools.dir/src/scheduled_thread_pool.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:106560: libs/core/thread_pools/CMakeFiles/hpx_thread_pools.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```


## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1. Compile HPX master using boost 17.6 and gcc 11.2.1

## Specifications

  - HPX Version: Master
  - Platform (compiler, OS): Fedora 35

--- a/libs/core/coroutines/include/hpx/coroutines/thread_id_type.hpp
+++ b/libs/core/coroutines/include/hpx/coroutines/thread_id_type.hpp
@@ -307,6 +307,16 @@ namespace hpx { namespace threads {
             thrd_.reset();
         }
 
+        void reset(thread_repr* thrd, bool add_ref = true) noexcept
+        {
+            thrd_.reset(thrd, add_ref);
+        }
+
+        constexpr thread_repr* detach() noexcept
+        {
+            return thrd_.detach();
+        }
+
         friend bool operator==(
             std::nullptr_t, thread_id_ref const& rhs) noexcept
         {

--- a/libs/core/schedulers/include/hpx/schedulers/thread_queue.hpp
+++ b/libs/core/schedulers/include/hpx/schedulers/thread_queue.hpp
@@ -111,7 +111,8 @@ namespace hpx { namespace threads { namespace policies {
         };
         using thread_description_ptr = thread_description*;
 #else
-        using thread_description_ptr = thread_id_ref_type;
+        using thread_description_ptr =
+            typename thread_id_ref_type::thread_repr*;
 #endif
 
         using work_items_type = typename PendingQueuing::template apply<
@@ -841,8 +842,10 @@ namespace hpx { namespace threads { namespace policies {
                 return true;
             }
 #else
-            if (0 != work_items_count && work_items_.pop(thrd, steal))
+            thread_description_ptr next_thrd;
+            if (0 != work_items_count && work_items_.pop(next_thrd, steal))
             {
+                thrd.reset(next_thrd, false);    // do not addref!
                 --work_items_count_.data_;
                 return true;
             }
@@ -860,7 +863,9 @@ namespace hpx { namespace threads { namespace policies {
                                  hpx::chrono::high_resolution_clock::now()},
                 other_end);
 #else
-            work_items_.push(HPX_MOVE(thrd), other_end);
+            // detach the thread from the id_ref without decrementing
+            // the reference count
+            work_items_.push(thrd.detach(), other_end);
 #endif
         }
 

